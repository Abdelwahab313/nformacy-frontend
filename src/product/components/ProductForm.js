import React from 'react';
import CssBaseline from '@material-ui/core/CssBaseline';
import Typography from '@material-ui/core/Typography';
import Grid from '@material-ui/core/Grid';
import TextField from '@material-ui/core/TextField';
import Button from '@material-ui/core/Button';
import Container from '@material-ui/core/Container';
import useForm from 'react-hook-form';
import useProductFormStyle from '../styles/productForm';

//TODO: review form validation
const ProductForm = ({
  pageTitle,
  onSubmit,
  name = '',
  price = '',
  sku = '',
}) => {
  const { register, handleSubmit, errors } = useForm();
  const classes = useProductFormStyle();
  return (
    <Container component='main' maxWidth='xs'>
      <CssBaseline />
      <div className={classes.paper}>
        <Typography component='h1' variant='h5'>
          {pageTitle}
        </Typography>
        <form
          onSubmit={handleSubmit(onSubmit)}
          className={classes.form}
          noValidate>
          <Grid container spacing={2}>
            <Grid item xs={6} sm={6} lg={12}>
              <TextField
                autoComplete='name'
                name='name'
                variant='outlined'
                fullWidth
                id={'productName'}
                inputProps={{ 'data-testid': 'productName' }}
                label='اسم المنتج'
                defaultValue={name}
                inputRef={register({ required: 'برجاء ادخال اسم المنتج' })}
                autoFocus
                error={errors.name}
                helperText={errors.name && errors.name.message}
              />
            </Grid>
            <Grid item xs={6} sm={6} lg={12}>
              <TextField
                autoComplete='name'
                name='sku'
                variant='outlined'
                fullWidth
                id={'sku'}
                inputProps={{ 'data-testid': 'sku' }}
                label='sku'
                defaultValue={sku}
                error={errors.sku}
                inputRef={register({ required: 'برجاء ادخال الرقم التسلسلى' })}
                helperText={errors.sku && errors.sku.message}
              />
            </Grid>
            <Grid item xs={6} sm={6} lg={12}>
              <TextField
                autoComplete='name'
                name='price'
                variant='outlined'
                fullWidth
                id={'price'}
                inputProps={{ 'data-testid': 'price' }}
                label='سعر المنتج'
                defaultValue={price}
                error={errors.price}
                inputRef={register({
                  required: 'برجاء ادخال سعر المنتج',
                  pattern: {
                    value: /^[+-]?(\d*\.)?\d+$/,
                    message: 'السعر يجب ان يكون رقم',
                  },
                })}
                helperText={errors.price && errors.price.message}
              />
            </Grid>
          </Grid>
          <Button
            type='submit'
            fullWidth
            variant='contained'
            id={'add_product_submit_btn'}
            color='primary'
            className={classes.submit}>
            حفظ
          </Button>
        </form>
      </div>
    </Container>
  );
};
export default ProductForm;
