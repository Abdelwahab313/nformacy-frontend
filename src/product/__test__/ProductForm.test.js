import React from 'react';
import { render } from '@testing-library/react';
import ProductForm from '../components/ProductForm';

describe('Product Form', () => {
  beforeEach(() => {
    return require('mutationobserver-shim');
  });
  it('should render add form with empty TextField if no props passed', () => {
    const onSubmit = jest.fn();
    let pageTitle = 'اضافه منتج جديد';
    const { asFragment } = render(
      <ProductForm pageTitle={pageTitle} onSubmit={onSubmit} />,
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it('should render add form with values if props passed', () => {
    const onSubmit = jest.fn();
    let pageTitle = 'اضافه منتج جديد';
    const { asFragment } = render(
      <ProductForm
        pageTitle={pageTitle}
        onSubmit={onSubmit}
        name={'test'}
        price={'123'}
        sku={'123'}
      />,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
