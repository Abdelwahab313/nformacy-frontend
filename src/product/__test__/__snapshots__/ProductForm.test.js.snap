// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product Form should render add form with empty TextField if no props passed 1`] = `
<DocumentFragment>
  <main
    class="MuiContainer-root MuiContainer-maxWidthXs"
  >
    <div
      class="makeStyles-paper-1"
    >
      <h1
        class="MuiTypography-root MuiTypography-h5"
      >
        اضافه منتج جديد
      </h1>
      <form
        class="makeStyles-form-2"
        novalidate=""
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-6 MuiGrid-grid-lg-12"
          >
            <div
              class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined Mui-focused Mui-focused"
                data-shrink="true"
                for="productName"
                id="productName-label"
              >
                اسم المنتج
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth Mui-focused Mui-focused MuiInputBase-formControl"
              >
                <input
                  aria-invalid="false"
                  autocomplete="name"
                  class="MuiInputBase-input MuiOutlinedInput-input"
                  data-testid="productName"
                  id="productName"
                  name="name"
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="PrivateNotchedOutline-root-204 MuiOutlinedInput-notchedOutline"
                  style="padding-left: 8px;"
                >
                  <legend
                    class="PrivateNotchedOutline-legend-205"
                    style="width: 0px;"
                  >
                    <span>
                      ​
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-6 MuiGrid-grid-lg-12"
          >
            <div
              class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
                data-shrink="false"
                for="sku"
                id="sku-label"
              >
                sku
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
              >
                <input
                  aria-invalid="false"
                  autocomplete="name"
                  class="MuiInputBase-input MuiOutlinedInput-input"
                  data-testid="sku"
                  id="sku"
                  name="sku"
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="PrivateNotchedOutline-root-204 MuiOutlinedInput-notchedOutline"
                  style="padding-left: 8px;"
                >
                  <legend
                    class="PrivateNotchedOutline-legend-205"
                    style="width: 0.01px;"
                  >
                    <span>
                      ​
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-6 MuiGrid-grid-lg-12"
          >
            <div
              class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
                data-shrink="false"
                for="price"
                id="price-label"
              >
                سعر المنتج
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
              >
                <input
                  aria-invalid="false"
                  autocomplete="name"
                  class="MuiInputBase-input MuiOutlinedInput-input"
                  data-testid="price"
                  id="price"
                  name="price"
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="PrivateNotchedOutline-root-204 MuiOutlinedInput-notchedOutline"
                  style="padding-left: 8px;"
                >
                  <legend
                    class="PrivateNotchedOutline-legend-205"
                    style="width: 0.01px;"
                  >
                    <span>
                      ​
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
        </div>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-submit-3 MuiButton-containedPrimary MuiButton-fullWidth"
          id="save_product_submit_btn"
          tabindex="0"
          type="submit"
        >
          <span
            class="MuiButton-label"
          >
            حفظ
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </form>
    </div>
  </main>
</DocumentFragment>
`;

exports[`Product Form should render add form with values if props passed 1`] = `
<DocumentFragment>
  <main
    class="MuiContainer-root MuiContainer-maxWidthXs"
  >
    <div
      class="makeStyles-paper-247"
    >
      <h1
        class="MuiTypography-root MuiTypography-h5"
      >
        اضافه منتج جديد
      </h1>
      <form
        class="makeStyles-form-248"
        novalidate=""
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-6 MuiGrid-grid-lg-12"
          >
            <div
              class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled Mui-focused Mui-focused"
                data-shrink="true"
                for="productName"
                id="productName-label"
              >
                اسم المنتج
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth Mui-focused Mui-focused MuiInputBase-formControl"
              >
                <input
                  aria-invalid="false"
                  autocomplete="name"
                  class="MuiInputBase-input MuiOutlinedInput-input"
                  data-testid="productName"
                  id="productName"
                  name="name"
                  type="text"
                  value="test"
                />
                <fieldset
                  aria-hidden="true"
                  class="PrivateNotchedOutline-root-450 MuiOutlinedInput-notchedOutline"
                  style="padding-left: 8px;"
                >
                  <legend
                    class="PrivateNotchedOutline-legend-451"
                    style="width: 0px;"
                  >
                    <span>
                      ​
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-6 MuiGrid-grid-lg-12"
          >
            <div
              class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled"
                data-shrink="true"
                for="sku"
                id="sku-label"
              >
                sku
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
              >
                <input
                  aria-invalid="false"
                  autocomplete="name"
                  class="MuiInputBase-input MuiOutlinedInput-input"
                  data-testid="sku"
                  id="sku"
                  name="sku"
                  type="text"
                  value="123"
                />
                <fieldset
                  aria-hidden="true"
                  class="PrivateNotchedOutline-root-450 MuiOutlinedInput-notchedOutline"
                  style="padding-left: 8px;"
                >
                  <legend
                    class="PrivateNotchedOutline-legend-451"
                    style="width: 0px;"
                  >
                    <span>
                      ​
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-6 MuiGrid-grid-lg-12"
          >
            <div
              class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled"
                data-shrink="true"
                for="price"
                id="price-label"
              >
                سعر المنتج
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
              >
                <input
                  aria-invalid="false"
                  autocomplete="name"
                  class="MuiInputBase-input MuiOutlinedInput-input"
                  data-testid="price"
                  id="price"
                  name="price"
                  type="text"
                  value="123"
                />
                <fieldset
                  aria-hidden="true"
                  class="PrivateNotchedOutline-root-450 MuiOutlinedInput-notchedOutline"
                  style="padding-left: 8px;"
                >
                  <legend
                    class="PrivateNotchedOutline-legend-451"
                    style="width: 0px;"
                  >
                    <span>
                      ​
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
        </div>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-submit-249 MuiButton-containedPrimary MuiButton-fullWidth"
          id="save_product_submit_btn"
          tabindex="0"
          type="submit"
        >
          <span
            class="MuiButton-label"
          >
            حفظ
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </form>
    </div>
  </main>
</DocumentFragment>
`;
